render/wgpu/src/target.rs

```diff
@@ -165,10 +165,19 @@ impl TextureTarget {
         }
     }
 
-    pub fn capture(&self, device: &wgpu::Device) -> Option<image::RgbaImage> {
-        let buffer_future = self.buffer.slice(..).map_async(wgpu::MapMode::Read);
-        device.poll(wgpu::Maintain::Wait);
-        match futures::executor::block_on(buffer_future) {
+    pub fn capture(&self) -> Option<image::RgbaImage> {
+        use std::sync::mpsc::channel;
+
+        let (sender, receiver) = channel();
+
+        self.buffer
+            .slice(..)
+            .map_async(wgpu::MapMode::Read, move |result| {
+                sender.send(result).unwrap()
+            });
+
+        let result = receiver.recv().unwrap();
```

Comment:
This will never complete: you need to interact with the `wgpu` runtime in order for the callback to run.
See https://docs.rs/wgpu/latest/wgpu/struct.BufferSlice.html#method.map_async